rest/route/host_spawn.go

```diff
@@ -1128,53 +1129,75 @@ func (h *hostRunScript) Factory() gimlet.RouteHandler {
 
 func (h *hostRunScript) Parse(ctx context.Context, r *http.Request) error {
 	var err error
-	h.hostID, err = validateID(gimlet.GetVars(r)["host_id"])
-	if err != nil {
-		return errors.Wrap(err, "invalid host id")
-	}
-
-	hostScript := model.APIHostScript{}
-	if err = util.ReadJSONInto(util.NewRequestReader(r), &hostScript); err != nil {
+	hostScriptOpts := model.APIHostScript{}
+	if err = util.ReadJSONInto(util.NewRequestReader(r), &hostScriptOpts); err != nil {
 		return errors.Wrap(err, "can't read host command from json")
 	}
-	h.script = hostScript.Script
+	h.script = hostScriptOpts.Script
+	h.hostIDs = hostScriptOpts.Hosts
 
 	return nil
 }
 
 func (h *hostRunScript) Run(ctx context.Context) gimlet.Responder {
-	user := MustHaveUser(ctx)
-	host, err := h.sc.FindHostByIdWithOwner(h.hostID, user)
-	if err != nil {
-		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "can't get host"))
-	}
-	if host.Status != evergreen.HostRunning {
-		return gimlet.MakeJSONInternalErrorResponder(errors.Errorf("can't run script on host with status '%s'", host.Status))
-	}
+	u := MustHaveUser(ctx)
+	type hostOutput struct {
+		hostID string
+		output string
+	}
+	hostsOutput := make(chan hostOutput, len(h.hostIDs))
+	wg := &sync.WaitGroup{}
+	wg.Add(len(h.hostIDs))
+	for _, hostID := range h.hostIDs {
+		go func(ctx context.Context, hostID string, u *user.DBUser) {
```

Comment:
I think the real problem is that there *is* a time out because the context that gets passed here is the one from the request, so commands have to complete within a minute, which isn't realistic.

What I was expecting here is that we'd start operations on jasper, and then return a list of jasper process IDs, and then let people pool for jasper process status seperatly. These requests should be non-blocking, though we can add a blocking-variant too, though the max timeout will need to be a lot less than our 1.minute timeout? (30s?)